src/main/java/com/redhat/rhjmc/containerjfr/net/HttpServer.java

```diff
@@ -19,18 +24,92 @@
+        String keystorePath = null;
+        if (env.hasEnv(KEYSTORE_PATH_ENV)) {
+            keystorePath = env.getEnv(KEYSTORE_PATH_ENV);
+        } else if (new File(System.getProperty("user.home"), "container-jfr-keystore.jks").exists()) {
+            keystorePath = System.getProperty("user.home") + File.separator + "container-jfr-keystore.jks";
+        } else if (new File(System.getProperty("user.home"), "container-jfr-keystore.pfx").exists()) {
+            keystorePath = System.getProperty("user.home") + File.separator + "container-jfr-keystore.pfx";
+        } else if (new File(System.getProperty("user.home"), "container-jfr-keystore.p12").exists()) {
+            keystorePath = System.getProperty("user.home") + File.separator + "container-jfr-keystore.p12";
+        }
+
+        String keystorePass = env.getEnv(KEYSTORE_PASS_ENV, "");
+        if (keystorePath != null) {
+            if (keystorePass.isEmpty()) {
+                logger.warn("keystore password unset or empty");
+            }
+
+            if (keystorePath.endsWith(".jks")) {
+                options.setSsl(true)
+                        .setKeyStoreOptions(new JksOptions().setPath(keystorePath).setPassword(keystorePass));
+                return true;
+            } else if (keystorePath.endsWith(".pfx") || keystorePath.endsWith(".p12")) {
+                options.setSsl(true)
+                        .setPfxKeyCertOptions(new PfxOptions().setPath(keystorePath).setPassword(keystorePass));
+                return true;
+            }
+
+            IllegalArgumentException e = new IllegalArgumentException("unrecognized keystore type");
+            logger.error(e);
+            throw e;
+        }
+
+        String keyPath = null;
+        if (env.hasEnv(KEY_PATH_ENV)) {
+            keyPath = env.getEnv(KEY_PATH_ENV);
+        } else if (new File(System.getProperty("user.home"), "container-jfr-key.pem").exists()) {
+            keyPath = System.getProperty("user.home") + File.separator + "container-jfr-key.pem";
+        }
+
+        String certPath = null;
+        if (env.hasEnv(CERT_PATH_ENV)) {
+            certPath = env.getEnv(CERT_PATH_ENV);
+        } else if (new File(System.getProperty("user.home"), "container-jfr-cert.pem").exists()) {
+            certPath = System.getProperty("user.home") + File.separator + "container-jfr-cert.pem";
+        }
+
+        if (keyPath != null && certPath != null) {
+            options.setSsl(true)
+                    .setPemKeyCertOptions(new PemKeyCertOptions().setKeyPath(keyPath).setCertPath(certPath));
+            return true;
+        } else if (keyPath != null ^ certPath != null) {
+            IllegalArgumentException e = new IllegalArgumentException("both a key and a certificate are required");
+            logger.error(e);
+            throw e;
+        }
+
+        logger.warn("SSL parameters not set. Fallback to plain HTTP.");
+        options.setSsl(false);
+        return false;
+    }
```

Comment:
Overall, this is quite a long method with a lot of branching logic. Can this be split into smaller methods, and tests added?

Reply:
Actually, this seems like it probably even warrants its own "SSLConfiguration" class or something.